objects/src/transaction/proven_tx.rs

```diff
@@ -130,17 +118,6 @@ impl ProvenTransaction {
     // --------------------------------------------------------------------------------------------
 
     fn validate(self) -> Result<Self, ProvenTransactionError> {
-        let known_output_ids = BTreeSet::from_iter(self.output_notes.iter().map(|n| n.note_id()));
-        let unknown_ids: Vec<_> = self
-            .output_note_details
-            .keys()
-            .filter(|&k| !known_output_ids.contains(k))
-            .cloned()
-            .collect();
-        if !unknown_ids.is_empty() {
```

Comment:
This is an example of improved type safety, the check is no longer necessary, because the data is no longer split into two places.